pingen/ir_attachment.py

```diff
@@ -1,123 +1,99 @@
-        document = attachment.pingen_document_ids[0] if attachment.pingen_document_ids else None
-        if attachment.send_to_pingen:
+        pingen_document_obj = self.env['pingen.document']
+        document = self.pingen_document_ids[0] if \
+            self.pingen_document_ids else None
+        if self.send_to_pingen:
             if document:
-                document.write({'state': 'pending'}, context=context)
+                document.write({'state': 'pending'})
             else:
                 pingen_document_obj.create(
-                    cr, uid,
-                    self._prepare_pingen_document_vals(
-                        cr, uid, attachment, context=context),
-                    context=context)
+                    self._prepare_pingen_document_vals())
         else:
             if document:
                 if document.state == 'pushed':
-                    raise osv.except_osv(
-                        _('Error'),
-                        _('The attachment %s is already pushed to pingen.com.') %
-                        attachment.name)
-                document.write({'state': 'canceled'}, context=context)
+                    raise UserError(
+                        _('Error. The attachment ' +
+                          '%s is already pushed to pingen.com.') %
+                        self.name)
+                document.write({'state': 'canceled'})
         return
 
-    def create(self, cr, uid, vals, context=None):
-        attachment_id = super(ir_attachment, self).create(cr, uid, vals, context=context)
+    def create(self, vals):
+        attachment_id = super(IrAttachment, self).create(vals)
         if 'send_to_pingen' in vals:
-            self._handle_pingen_document(cr, uid, attachment_id, context=context)
+            self._handle_pingen_document(attachment_id)
         return attachment_id
 
-    def write(self, cr, uid, ids, vals, context=None):
-        res = super(ir_attachment, self).write(cr, uid, ids, vals, context=context)
+    def write(self, vals):
+        res = super(IrAttachment, self).write(vals)
         if 'send_to_pingen' in vals:
-            for attachment_id in ids:
-                self._handle_pingen_document(cr, uid, attachment_id, context=context)
+            for attachment_id in self:
+                self._handle_pingen_document(attachment_id)
         return res
 
-    def _decoded_content(self, cr, uid, attachment, context=None):
+    def _decoded_content(self):
         """ Returns the decoded content of an attachment (stored or url)
-
         Returns None if the type is 'url' and the url is not reachable.
         """
         decoded_document = None
-        if attachment.type == 'binary':
-            decoded_document = base64.decodestring(attachment.datas)
-        elif attachment.type == 'url':
-            response = requests.get(attachment.url)
+        if self.type == 'binary':
+            decoded_document = self.datas
```

Comment:
Can you explain why this should no longer be decoded? BTW the method name is `_decoded_content` so it isn't consistent.

Reply:
here file is decoded:
https://github.com/OCA/report-print-send/pull/114/files#diff-ea036c00459d0303eb70c0c27049197eR148